src/cloudflare/internal/vectorize-api.ts

```diff
@@ -32,24 +33,26 @@ function toNdJson(arr: object[]): string {
  * and not visible to end users.
  */
 class VectorizeIndexImpl implements Vectorize {
+  #fetcher: Fetcher;
+  #indexId: string;
+  #indexVersion: VectorizeVersion;
+  #useNdJson: boolean;
+
   constructor(
-    // TODO(soon): Can we use the # syntax here?
-    // eslint-disable-next-line no-restricted-syntax
-    private readonly fetcher: Fetcher,
-    // TODO(soon): Can we use the # syntax here?
-    // eslint-disable-next-line no-restricted-syntax
-    private readonly indexId: string,
-    // TODO(soon): Can we use the # syntax here?
-    // eslint-disable-next-line no-restricted-syntax
-    private readonly indexVersion: VectorizeVersion,
-    // TODO(soon): Can we use the # syntax here?
-    // eslint-disable-next-line no-restricted-syntax
-    private readonly useNdJson: boolean
-  ) {}
+    fetcher: Fetcher,
+    indexId: string,
+    indexVersion: VectorizeVersion,
+    useNdJson: boolean
+  ) {
+    this.#fetcher = fetcher;
+    this.#indexId = indexId;
+    this.#indexVersion = indexVersion;
+    this.#useNdJson = useNdJson;
+  }
```

Comment:
Let's have the vectorize-api team do a review...

Reply:
Given the enum changes, etc I still think we need the vectorize-api team to review.

Reply:
Resolving. Vectorize team reviewed these changes - https://chat.google.com/room/AAAABGzNtz4/KqAZj2eT5dQ/7vsvz4NOQpo?cls=10